Collector/Stack.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+final class Stack
+{
+    /**
+     * @var string
+     */
+    private $client;
+
+    /**
+     * @var Profile[]
+     */
+    private $profiles = [];
+
+    /**
+     * @var string
+     */
+    private $request;
+
+    /**
+     * @var string
+     */
+    private $response;
+
+    /**
+     * @var bool
+     */
+    private $failed = false;
+
+    /**
+     * @return string
+     */
+    public function getClient()
+    {
+        return $this->client;
+    }
+
+    /**
+     * @param string $client
+     */
+    public function setClient($client)
```

Comment:
same here, i think this and setRequest should be mandatory constructor arguments